### PR TITLE
Binary draw

### PR DESCRIPTION
**Pull request by Iaxama**
This PR adds a binary drawer that outputs a monochrome image. 
Output port of vFramer changed to BufferedPort<FlexiImage> to handle both monochrome and RGB outputs. 
vPreProcess has an additional feature to split polarities in two separate ports.

Completely retrocompatible.

### CODE REVIEW

**Comment by arrenglover** [comment]
@Iaxama please resolve the conflicts first before we do the review. it should just require a merge of master into your branch.

**Comment by Iaxama** [comment]
I have fixed all the conflicts

File: src/processing/vPreProcess/src/vPreProcess.cpp [hunk 1]
```diff
@@ -100,6 +100,8 @@ bool vPreProcess::configure(yarp::os::ResourceFinder &rf)
             rf.check("precheck", Value(true)).asBool();
     split_stereo = rf.check("split_stereo") &&
             rf.check("split_stereo", Value(true)).asBool();
+    split_polarities = rf.check("split_polarities") &&
+                       rf.check("split_polarities", Value(true)).asBool();
```
Comment (arrenglover) [inline src/processing/vPreProcess/src/vPreProcess.cpp]: check indentation here

File: src/processing/vPreProcess/src/vPreProcess.cpp [hunk 2]
```diff
@@ -364,25 +388,47 @@ void vPreProcess::run()
...
+                        if (split_polarities){
+                            if (v.polarity){
+                                qleft_pos.push_back(v);
+                            } else {
+                                qleft_neg.push_back(v);
+                            }
+                        } else {
                             qleft.push_back(v);
-                    }
+                        }                    }
```
Comment (arrenglover) [inline src/processing/vPreProcess/src/vPreProcess.cpp]: check indentation / brackets here

**Comment by arrenglover** [comment]
@Iaxama double check the indentations/brackets are correct and the merge is good to go

**Comment by arrenglover** [comment]
@Iaxama - now there are spaces between if statements and the first opening brackets. Can you revert it back to the previous formatting style.

**Comment by Iaxama** [comment]
> @Iaxama - now there are spaces between if statements and the first opening brackets. Can you revert it back to the previous formatting style.

Formatting style was not consistent across the file. Now I have reformatted removing the space after the if keyword.

**Pull request merged**